### PR TITLE
library: Some improvements to the Launcher demo

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

File: src/Library/demos/Launcher/main.blp [hunk 1]
```diff
@@ -23,73 +23,72 @@ Adw.StatusPage {
         Gtk.StackPage {
           name: "file_launcher";
           title: _("File Launcher");
-          child:
+          child: Box {
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.blp:26]: this is in our style guide :wink: 

https://github.com/sonnyp/Workbench/wiki/Style-Guide

File: src/Library/demos/Launcher/main.blp [hunk 2]
```diff
@@ -23,73 +23,72 @@ Adw.StatusPage {
...
-                    label: _("workbench.txt");
-                    styles ["dim-label", "body"]
-                  }
-                };
-              }
-              LinkButton {
-                label: _("API Reference");
-                uri : "https://docs.gtk.org/gtk4/class.FileLauncher.html";
+              Button file_location {
+                label: _("View in Files");
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.blp:41]: renamed from "View" to "View in Files" which is the recommended label for this

File: src/Library/demos/Launcher/main.js [hunk 3]
```diff
@@ -3,14 +3,6 @@ import Gdk from "gi://Gdk";
 import Gio from "gi://Gio";
 import GLib from "gi://GLib";
 
-const launch_file = workbench.builder.get_object("launch_file");
-const file_name = workbench.builder.get_object("file_name");
-const file_location = workbench.builder.get_object("file_location");
-const change_file = workbench.builder.get_object("change_file");
-const uri_launch = workbench.builder.get_object("uri_launch");
-const uri_details = workbench.builder.get_object("uri_details");
-const change_uri = workbench.builder.get_object("change_uri");
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.js:12]: the `Gio._promisify` should be directly below the import statements as they related to them

File: src/Library/demos/Launcher/main.js [hunk 4]
```diff
@@ -20,59 +12,56 @@ Gio._promisify(
 Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
 Gio._promisify(Gtk.UriLauncher.prototype, "launch", "launch_finish");
 
-const file_launcher = new Gtk.FileLauncher();
-const uri_launcher = new Gtk.UriLauncher();
-const file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
-  "Library/demos/Launcher/workbench.txt",
-);
-file_launcher.set_file(file);
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.js:28]: please use properties instead of methods

File: src/Library/demos/Launcher/main.js [hunk 5]
```diff
@@ -20,59 +12,56 @@ Gio._promisify(
...
-file_launcher.always_ask = true;
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.js:29]: we can set properties directly when instanciating see below `new Gtk.FileLauncher({`

File: src/Library/demos/Launcher/main.js [hunk 6]
```diff
@@ -20,59 +12,56 @@ Gio._promisify(
...
-    modal: true,
-  });
-  dialog_for_file
+  new Gtk.FileDialog()
     .open(workbench.window, null)
-    .then((result) => {
-      new_file = result;
-    })
-    .catch((err) => {
-      console.debug(err.message);
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.js:63]: don't use `console.debug` please, it's silent by default

all demos use `logError`

File: src/Library/demos/Launcher/main.js [hunk 7]
```diff
@@ -20,59 +12,56 @@ Gio._promisify(
...
+    .then((file) => {
+      file_launcher.file = file;
     })
-    .finally(() => {
-      file_launcher.set_file(new_file);
```
Comment (sonnyp) [inline src/Library/demos/Launcher/main.js:66]: please use property getters/setters, but also, we don't want to do this when there is an error

File: src/Library/demos/Launcher/main.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "Launcher",
-  "category": "controls",
-  "description": "Collect the arguments that are needed to open a File or URI",
+  "category": "platform",
+  "description": "Open a File or URI",
```
Comment (andyholmes) [inline src/Library/demos/Launcher/main.json:4]: Where was this string from, by the way?

File: src/Library/demos/Launcher/main.blp (hunk 2 above)
Comment (andyholmes) [reply src/Library/demos/Launcher/main.blp:41]: Sorry, where is this recommendation found?

File: src/Library/demos/Launcher/main.js (hunk 3 above)
Comment (andyholmes) [reply src/Library/demos/Launcher/main.js:12]: Added to the style guide.

File: src/Library/demos/Launcher/main.blp (hunk 2 above)
Comment (sonnyp) [reply src/Library/demos/Launcher/main.blp:41]: I asked Tobias a while ago :p

File: src/Library/demos/Launcher/main.json (hunk 8 above)
Comment (sonnyp) [reply src/Library/demos/Launcher/main.json:4]: https://docs.gtk.org/gtk4/class.UriLauncher.html

File: src/Library/demos/Launcher/main.js (hunk 5 above)
Comment (andyholmes) [reply src/Library/demos/Launcher/main.js:29]: Also added to Style Guide for future reference :)

File: src/Library/demos/Launcher/main.json (hunk 8 above)
Comment (andyholmes) [reply src/Library/demos/Launcher/main.json:4]: I mean the string you changed it to?

**Review by SoNiC-HeRE — APPROVED** [review]
Looks good to me; I'll take a closer look at the modifications and incorporate them into my upcoming PRs.

**Pull request merged**